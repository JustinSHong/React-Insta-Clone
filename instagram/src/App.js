import React, { Component } from "react";
import { render } from "react-dom";
import logo from "./logo.svg";
import "./App.css";
import dummyData from "./dummy-data";
// import SearchBar from './SearchBar/SearchBar.js';
import PostContainer from "./components/PostContainer/PostContainer.js";
import CommentSection from "./components/CommentSection/CommentSection.js";

class App extends Component {
  constructor() {
    super();
    this.state = {
      dummyList: []
    };
  }

  componentDidMount = () => {
    this.setState({ dummyList: dummyData });
    // console.log(dummyData);
  };

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        {/* <SearchBar /> */}
        <PostContainer dummy={this.state.dummyList} />
        {/* <CommentSection /> */}
      </div>
    );
  }
}

export default App;
render(<App />, document.getElementById("root"));
