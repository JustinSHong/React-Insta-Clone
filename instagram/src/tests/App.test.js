import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
// data
import postData from "../dummy-data";
// component
import App from "../App";

Enzyme.configure({ adapter: new Adapter() });

describe("<App />", () => {
	it("renders correctly", () => {
		const tree = renderer.create(<App />);
		expect(tree).toMatchSnapshot();
	});

	it("calls componentDidMount", () => {
		const spy = jest.spyOn(App.prototype, "componentDidMount");
		const component = mount(<App />);
		expect(spy).toHaveBeenCalled();
	});
});
