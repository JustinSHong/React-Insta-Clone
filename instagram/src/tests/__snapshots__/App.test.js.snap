// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="SearchBar"
  >
    <div
      className="brand"
    >
      <a
        href="https://www.instagram.com/?hl=en"
      >
        <i
          className="fab fa-instagram fa-3x"
        />
      </a>
      <p>
        instagram
      </p>
    </div>
    <form
      onSubmit={[Function]}
    >
      <input
        name="user"
        onChange={[Function]}
        placeholder="Search here..."
        type="text"
        value=""
      />
    </form>
    <div>
      <a
        className="accentLogo"
        href="https://www.instagram.com/?hl=en"
      >
        <i
          className="far fa-compass fa-2x"
        />
      </a>
      <a
        className="accentLogo"
        href="https://www.instagram.com/?hl=en"
      >
        <i
          className="far fa-heart fa-2x"
        />
      </a>
      <a
        className="accentLogo"
        href="https://www.instagram.com/?hl=en"
      >
        <i
          className="far fa-user fa-2x"
        />
      </a>
    </div>
  </div>
  <div
    className="postWrapper"
  >
    <div
      className="card card"
    >
      <div
        className="cardHeader"
      >
        <img
          alt="thumbnail logo"
          src="https://instagram.com/p/BkIOT12B87B/media/?size=m"
        />
        <h5
          className="card-title"
        >
          philzcoffee
        </h5>
      </div>
      <img
        className="cardImg card-img"
        placeholder="philzcoffee"
        src="https://instagram.com/p/Bl8TDqYBhNW/media/?size=l"
      />
      <div
        className="cardBody card-body"
      >
        <div>
           
          <i
            className="far fa-heart"
          />
          <i
            className="far fa-comment"
          />
        </div>
        <div>
          <strong>
            400
             likes
          </strong>
        </div>
        <div>
          <div>
            <strong>
              philzcoffee
            </strong>
             
            We've got more than just delicious coffees to offer at our shops!
          </div>
          <div>
            <strong>
              biancasaurus
            </strong>
             
            Looks delicious!
          </div>
          <div>
            <strong>
              martinseludo
            </strong>
             
            Can't wait to try it!
          </div>
          <div>
            <p
              className="timeStamp"
            >
              <span>
                Today at 4:46 PM
              </span>
            </p>
            <hr />
            <form
              className="CommentSection_form"
            >
              <input
                className="CommentSection_input border-top-0 border-right-0 border-left-0 border-bottom-0 py-4"
                name="newComment"
                onChange={[Function]}
                onClick={[Function]}
                placeholder="Add a comment..."
                type="text"
                value=""
              />
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card card"
    >
      <div
        className="cardHeader"
      >
        <img
          alt="thumbnail logo"
          src="https://scontent-lax3-1.cdninstagram.com/vp/8e4d5751a9757258bf4082eadf9b45e9/5BFB8940/t51.2885-19/10684037_1512272482349821_444823433_a.jpg?_nc_eui2=AeF2FJP_Ix-7FL2MpffxgiS_McvWO7x67X8Yezr4DNmjFI1bL1qMwCbFsT9YqBxD0I3nf1ju0P74omQ7BPMYtR_7"
        />
        <h5
          className="card-title"
        >
          twitch
        </h5>
      </div>
      <img
        className="cardImg card-img"
        placeholder="twitch"
        src="https://www.instagram.com/p/Bj_A97xnvwe/media/?size=l"
      />
      <div
        className="cardBody card-body"
      >
        <div>
           
          <i
            className="far fa-heart"
          />
          <i
            className="far fa-comment"
          />
        </div>
        <div>
          <strong>
            4307
             likes
          </strong>
        </div>
        <div>
          <div>
            <strong>
              twitch
            </strong>
             
            Epic Street Fighter action here in Las Vegas at #EVO2017!
          </div>
          <div>
            <strong>
              michaelmarzetta
            </strong>
             
            Omg that match was crazy
          </div>
          <div>
            <strong>
              themexican_leprechaun
            </strong>
             
            What a setup
          </div>
          <div>
            <strong>
              dennis_futbol
            </strong>
             
            It that injustice
          </div>
          <div>
            <strong>
              dennis_futbol
            </strong>
             
            Is
          </div>
          <div>
            <p
              className="timeStamp"
            >
              <span>
                Today at 4:46 PM
              </span>
            </p>
            <hr />
            <form
              className="CommentSection_form"
            >
              <input
                className="CommentSection_input border-top-0 border-right-0 border-left-0 border-bottom-0 py-4"
                name="newComment"
                onChange={[Function]}
                onClick={[Function]}
                placeholder="Add a comment..."
                type="text"
                value=""
              />
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card card"
    >
      <div
        className="cardHeader"
      >
        <img
          alt="thumbnail logo"
          src="https://scontent-lax3-1.cdninstagram.com/vp/19e6a25a00cc1eea07dc4b79d7438661/5C06F201/t51.2885-19/s320x320/13398400_140638743023210_310840336_a.jpg?_nc_eui2=AeHaTO7isTOyfHGJnQjzXCirCt-m-Ba5MJj_DaTodZDkykxamGEIwMYnraO_UMF0DVCclr8cw7Do55gan_zEir_R"
        />
        <h5
          className="card-title"
        >
          playhearthstone
        </h5>
      </div>
      <img
        className="cardImg card-img"
        placeholder="playhearthstone"
        src="https://www.instagram.com/p/BjNCjjfFe7z/media/?size=l"
      />
      <div
        className="cardBody card-body"
      >
        <div>
           
          <i
            className="far fa-heart"
          />
          <i
            className="far fa-comment"
          />
        </div>
        <div>
          <strong>
            5306
             likes
          </strong>
        </div>
        <div>
          <div>
            <strong>
              playhearthstone
            </strong>
             
            Power alone is not to be feared. Fear instead those who wield it! #FrozenThrone #Expansion #DeathKnights
          </div>
          <div>
            <strong>
              tapmelon
            </strong>
             
            Wish that death knight could be added as a new playable class in this expansion.
          </div>
          <div>
            <strong>
              micpetboudreau
            </strong>
             
            Can't wait
          </div>
          <div>
            <strong>
              awaywetravel
            </strong>
             
            I &lt;3 Hearthstone.
          </div>
          <div>
            <strong>
              awesomebt28
            </strong>
             
            I like how gul'dan looks so old and useless
          </div>
          <div>
            <p
              className="timeStamp"
            >
              <span>
                Today at 4:46 PM
              </span>
            </p>
            <hr />
            <form
              className="CommentSection_form"
            >
              <input
                className="CommentSection_input border-top-0 border-right-0 border-left-0 border-bottom-0 py-4"
                name="newComment"
                onChange={[Function]}
                onClick={[Function]}
                placeholder="Add a comment..."
                type="text"
                value=""
              />
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
